src/modm/platform/uart/stm32/uart_hal_impl.hpp.in

```diff
@@ -28,45 +35,56 @@ namespace modm::platform
 
 // ----------------------------------------------------------------------------
 void
-{{ name }}::enable()
+{{ hal }}::enable()
 {
-	Rcc::enable<Peripheral::{{ uart_name ~ id }}>();
+	Rcc::enable<Peripheral::{{ name }}>();
 }
 
 void
-{{ name }}::disable()
+{{ hal }}::disable()
 {
 	// TX, RX, Uart, etc. Disable
 	{{ peripheral }}->CR1 = 0;
-	Rcc::disable<Peripheral::{{ uart_name ~ id }}>();
+	Rcc::disable<Peripheral::{{ name }}>();
 }
 
 void
-{{ name }}::enableOperation()
+{{ hal }}::enableOperation()
 {
 	{{ peripheral }}->CR1 |= USART_CR1_UE;
 }
 
 void
-{{ name }}::disableOperation()
+{{ hal }}::disableOperation()
 {
 	{{ peripheral }}->CR1 &= ~USART_CR1_UE;
 }
 
 template< class SystemClock, modm::baudrate_t baudrate, modm::percent_t tolerance >
 void
-{{ name }}::initialize(Parity parity, WordLength length)
+{{ hal }}::initialize(Parity parity, WordLength length)
 {
 	enable();
 	disableOperation();
 
 %% if over8
-	constexpr uint32_t scalar = (baudrate * 16l > SystemClock::{{ uart_name ~ id }}) ? 8 : 16;
+	constexpr uint32_t scalar = (baudrate * 16l > SystemClock::{{ name }}) ? 8 : 16;
 	constexpr uint32_t max = ((scalar == 16) ? (1ul << 16) : (1ul << 15)) - 1ul;
+	constexpr auto result = Prescaler::from_range(SystemClock::{{ name }}, baudrate, 1, max);
+%% elif uart_type == "Lpuart"
+	static_assert(SystemClock::{{ name }} > baudrate * 3,
+		"fck must be in the range [3 x baud rate, 4096 x baud rate].");
+	static_assert(SystemClock::{{ name }} < baudrate * 4096,
```

Comment:
Isn't it an inclusive range, so <= and >= ?

Reply:
Also overflow again :wink: , `baudrate_t` is `uint32_t`

Reply:
For > 1Mbit/s Uart `baudrate * 4096` can overflow

Reply:
Your right!

![grafik](https://user-images.githubusercontent.com/2820734/117590820-7e92dd00-b131-11eb-9e88-c7f12c833a5f.png)